### ISSUE TITLE
Issue using <View> with renderLink 

### ISSUE THREAD

**Issue by tomeberle** [joshswan/react-native-autolink#48]
Hi there, 

I am trying to create a newsfeed with automatic links previewing that would look like the following example from react-native-url-preview.

![Screenshot 2020-03-19 at 17 27 39](https://user-images.githubusercontent.com/37332240/77090066-f02c3b00-6a06-11ea-84fa-61f69264252a.png)

In order to detect URLs within the text of each posts, I am using this great autolink library. Then, I hopped to use 'renderLink' prop to design and render the preview of the link. 

However, there is a limitation from the library as renderLink can only return a Text component. So when I try to add a view to renderLink like the following: 

`renderLink={(text, match, index) =>
                          (<View><Text>{text}</Text></View)}`

I get those funny file icons: 
![Screenshot 2020-03-19 at 17 36 53](https://user-images.githubusercontent.com/37332240/77090978-39c95580-6a08-11ea-8fd0-bf4bc3419b5d.png)

Would someone see anyway I could tackle this issue ? 

Thanks for the help!

**Comment by joshswan** [comment]
Definitely happy to see if there's something we can do to make this work as using more complex elements within the linked text output is definitely something that's come up before.

Currently, Autolink wraps the entire output in a single Text component, which is why generally just outputting a View using renderLink is problematic. RN60 introduced some limited support for inline Views within Text, but it's still only partially supported as of RN 61. I just played around with it and while I could get a View to display nicely within the output, all the text within that View was invisible - not exactly ideal. I'd imagine support will improve though.

Thinking about it, we could make the wrapping component a prop so that you can choose to have Autolink wrap the output in a View rather than Text. I'm struggling to figure out how to handle the style and other Text-related props which currently get passed through to that root Text node though.

```javascript
<Autolink
  text="some text to link..."
  component={View}
  // this would work now
  renderLink={(text) => <View><Text>{text}</Text></View>}
  // not sure how to handle these without massive breaking changes
  style={someStyles}
  numberOfLines={2}
/>
```

**Comment by joshswan** [comment]
Just released v3.0.0 with support for customizing the container component via the `component` prop, and just about everything else with the new `linkProps`, `textProps`, and `renderText` props.

To get you started, something like the following should now be possible:
```js
<Autolink
  text={text}
  component={View}
  renderLink={(text, match, index) => <LinkPreviewComponent />}
  // Might need to remove leading spaces after block elements
  renderText={(text, index) => <Text>{text.trim()}</Text>
/>
```

**Comment by tomeberle** [comment]
Thanks a lot @joshswan for this fast change. I'll try it right away.
